### PR TITLE
[MINOR][PYTHON][DOCS] Remove duplicated versionchanged per versionadded

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR addresses all the cases of duplicated `versionchanged` directives with `versionadded` directives, see also https://github.com/apache/spark/pull/42597.

Also, this PR mentions that all functions support Spark Connect from Apache Spark 3.5.0.

### Why are the changes needed?

To remove duplicated information in docstring.

### Does this PR introduce _any_ user-facing change?

Yes, it removes duplicated information in PySpark API Reference page.

### How was this patch tested?

CI in this PR should validate them.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @LuciferYang @yaooqinn @zhengruifeng FYI

File: python/pyspark/sql/conf.py [hunk 1]
```diff
@@ -39,11 +39,7 @@ def __init__(self, jconf: JavaObject) -> None:
 
     @since(2.0)
     def set(self, key: str, value: Union[str, int, bool]) -> None:
-        """Sets the given Spark runtime configuration property.
-
-        .. versionchanged:: 3.4.0
```
Comment (HyukjinKwon) [inline python/pyspark/sql/conf.py:44]: ` .. versionchanged:: 3.4.0` is mentioned at the class level above.

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -626,6 +623,10 @@ def mode(col: "ColumnOrName") -> Column:
     :class:`~pyspark.sql.Column`
         the most frequent value in a group.
 
+    Notes
```
Comment (zhengruifeng) [inline python/pyspark/sql/functions.py:626]: I am fine with this `Notes`

but does it means all the 150+ functions added in 3.5.0 needs this?

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py:626]: That's a good point..

File: python/pyspark/sql/conf.py [hunk 3]
```diff
@@ -66,11 +59,7 @@ def get(
 
     @since(2.0)
     def unset(self, key: str) -> None:
-        """Resets the configuration property for the given key.
-
-        .. versionchanged:: 3.4.0
-            Supports Spark Connect.
-        """
+        """Resets the configuration property for the given key."""
```
Comment (yaooqinn) [inline python/pyspark/sql/conf.py:62]: This comment is not related to the current PR. However, I have a question about the accuracy of this document. Unsets?

Comment (HyukjinKwon) [reply python/pyspark/sql/conf.py:62]: Lol let's fix it since we're here.

Comment (HyukjinKwon) [reply python/pyspark/sql/conf.py:62]: Actually, I think it's fine. It will reset it to its default value .. in a way :-).

Comment (HyukjinKwon) [reply python/pyspark/sql/conf.py:62]: Scala side has the same doc so let's leave it here for now.

Comment (yaooqinn) [reply python/pyspark/sql/conf.py:62]: Yes, I am okay with it.

**Review by yaooqinn — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.
